### ISSUE TITLE
HTTP request with Http Authenticated

### ISSUE THREAD

**Issue by Dzegaib** [Codeception/Codeception#2412]
Hello, i'm trying to do HTTP request with Authtenticate, but always return "401". Follow my steps:
```php
$I->wantTo('Login on API');
$I->amHttpAuthenticated('test1', 'test1');
$I->haveHttpHeader('Content-Type', 'application/x-www-form-urlencoded');
$I->sendGET('/?username=usertest1&password=ABC123test');
$I->seeResponseCodeIs(200);
```

Follow my api.suite.yml:
```yaml
class_name: ApiTester
modules:
    enabled:
      - REST:
          url: 'https://test_api/login'
          auth: ['test1','test1']
          depends: PhpBrowser
          part: Json
```
Someone, can help me please? 

Tks

**Comment by DavertMik** [comment]
Sorry it's impossible to follow your steps: 

* we know nothing of the app you are testing
* your config is not formatted, probably your issue is missing spaces. Yaml reqiures 4 spaces to be strictly used for intendation

**Comment by Naktibalda** [comment]
@DavertMik  Formatting was correct, I added triple backticks to make Github keep formatting.

@Dzegaib why do you use 3 different ways to pass login credentials in one test?
REST module does not support auth parameter in configuration, but PhpBrowser does.

Run your tests with `-vv` flag to get more information about request and response.

**Comment by Dzegaib** [comment]
@Naktibalda Hi!!

The application, that we have here, use Digest do Authtenticate, the first Authenticate is to API, the lats is just to do my login. So now i'm using the method "amDigestAuthenticated", but i don't now what is happen, I use "-vv" and i could see this:

[Request headers] {"Content-Type":"application/x-www-form-urlencoded"}
  [Request] GET https://test_api/login?username=usertest1&password=ABC123test
  [Response] {"error":true,"message":"Authentication required.","code":"0403"}
  [Headers] {"Server":["nginx"],"Date":["Tue, 29 Sep 2015 18:35:11 GMT"],"Content-Type":["application/json"],"Transfer-Encoding":["chunked"],"Connection":["close"],"Www-Authenticate":["Digest realm="API Dafiti", domain="/category", nonce="395af5baa9c2263520fcaa0517317474", opaque="d75db7b160fe72d1346d2bd1f67bfd10", algorithm="MD5", qop="auth""],"Set-Cookie":["BIGipServerpool_api_forward_http=2483202240.24064.0000; path=/"]}
  [Status] 401

I think that my Header have some problem, but i don't know how to config my Header to send a HTTP request with Digest Authenticate.

Tks

**Comment by Naktibalda** [comment]
Do you get the same response if you use incorrect password in `amDigestAuthenticated`?

**Comment by Naktibalda** [comment]
@Dzegaib Have you solved this issue?

**Comment by doctorx** [comment]
This is the bug with amDigestAuthenticated.
Config:
```yaml
class_name: ApiTester
modules:
    enabled:
        - \Helper\Api
        - REST:
            depends: PhpBrowser
            url: http://hostname/
            part: Json
```
Code: 
```php
$I->wantTo('create new order for not register member');
$I->amDigestAuthenticated($user, $password);
$I->sendGET($uri);
$I->seeResponseCodeIs(401);
```
But do not send auth digest request header on server:
Bug in here:
*codecept.phar/src/Codeception/Module/REST.php:268*
We use Grizzle6 client where call method with signature **setAuth($username, $passwrod)**
But now wrong call  **$this->client->setAuth($username, $password, CURLAUTH_DIGEST);**

**Comment by doctorx** [comment]
Hmm. Commit is broken.

**Comment by Naktibalda** [comment]
Digest authentication was fixed in 2.2.4 #3416
